core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -162,6 +162,7 @@ class Partition(val topic: String,
    */
   def makeLeader(controllerId: Int,
                  partitionStateInfo: PartitionStateInfo, correlationId: Int): Boolean = {
+    var leaderHWIncremented = false
```

Comment:
Thanks for making the change. To be honest, I am not drinking the Scala kool-aid either. It's my preferred language in the JVM, but it has plenty of warts.